### PR TITLE
Feature/spliceai

### PR DESCRIPTION
**Pull request by jemten**
### This PR fixes:

- adds annotation from spliceai
- updates rankmodel config with spliceai max delta score 
- adds recipe to download vcfanno functions from reference repo

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: documentation/Setup.md [hunk 1]
```diff
@@ -139,6 +139,12 @@ Corresponding MIP references:
 ## Gene panel for the clinical test
 MIP will split the variants into two sets (clinical a.k.a "selected" and research) based on gene coordinates and hgnc_id, which is recorded in gene_panels.bed file(s) using MIPs own vcfparser. A template for grch37 can be found in MIPs dir under `templates/gene_panels.bed`.  
 
+#### SpliceAI - Splicing predicition
```
Comment (henrikstranneheim) [inline documentation/Setup.md:142]: Nice!

File: lib/MIP/Recipes/Analysis/Variant_annotation.pm [hunk 2]
```diff
@@ -532,6 +532,7 @@ sub analysis_variant_annotation_panel {
             filehandle           => $filehandle,
             infile_path          => $infile_path,
             toml_configfile_path => $active_parameter_href->{vta_vcfanno_config},
+            luafile_path         => $active_parameter_href->{vta_vcfanno_functions},
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Variant_annotation.pm:535]: Order AB

File: lib/MIP/Recipes/Pipeline/Download_rd_dna.pm [hunk 3]
```diff
@@ -165,6 +166,7 @@ sub pipeline_download_rd_dna {
         sv_vta_vcfanno_config          => \&download_sv_vta_vcfanno_config,
         vcf2cytosure_blacklist_regions => \&download_vcf2cytosure_blacklist_regions,
         vta_vcfanno_config             => \&download_vta_vcfanno_config,
+        vcfanno_functions              => \&download_vcfanno_functions,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Pipeline/Download_rd_dna.pm:169]: Order AB

File: lib/MIP/Recipes/Pipeline/Download_rd_dna.pm [hunk 4]
```diff
@@ -122,6 +122,7 @@ sub pipeline_download_rd_dna {
     use MIP::Recipes::Download::Vcf2cytosure_blacklist_regions
       qw{ download_vcf2cytosure_blacklist_regions };
     use MIP::Recipes::Download::Vta_vcfanno_config qw{ download_vta_vcfanno_config };
+    use MIP::Recipes::Download::Vcfanno_functions qw{ download_vcfanno_functions };
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Pipeline/Download_rd_dna.pm:125]: Order AB

File: lib/MIP/Vcfanno.pm [hunk 5]
```diff
@@ -109,11 +119,29 @@ sub check_vcfanno_toml {
             {
                 object_name    => q{file},
                 object_type    => q{file},
-                parameter_name => $parameter_name,
+                parameter_name => $parameter_names_ref->[0],
```
Comment (henrikstranneheim) [inline lib/MIP/Vcfanno.pm:122]: Could we unpack this as:
my $vcfanno_config = $parameter_names_ref->[0];
before this to make it more clear what is expected?

File: lib/MIP/Vcfanno.pm [hunk 6]
```diff
@@ -109,11 +119,29 @@ sub check_vcfanno_toml {
...
+        check_filesystem_objects_and_index_existance(
+            {
+                object_name    => q{file},
+                object_type    => q{file},
+                parameter_name => $parameter_names_ref->[1],
+                path           => $vcfanno_config{functions}{file},
+            }
+        );
+        ## Set path in active parameter
+        $active_parameter_href->{ $parameter_names_ref->[1] } =
```
Comment (henrikstranneheim) [inline lib/MIP/Vcfanno.pm:141]: Could we unpack this as:
my $vcfanno_function = $parameter_names_ref->[1];
before this to make it more clear what is expected?

File: lib/MIP/Vcfanno.pm [hunk 7]
```diff
@@ -142,19 +170,21 @@ sub parse_toml_config_parameters {
...
         next CONFIG_FILE if ( not $active_parameter_href->{$recipe_name} );
 
         check_vcfanno_toml(
             {
-                parameter_name    => $parameter_name,
-                vcfanno_file_toml => $active_parameter_href->{$parameter_name},
+                active_parameter_href => $active_parameter_href,
+                parameter_names_ref   => $parameter_names_ref,
+                vcfanno_file_toml =>
+                  $active_parameter_href->{ $parameter_names_ref->[0] },
```
Comment (henrikstranneheim) [inline lib/MIP/Vcfanno.pm:187]: Same here

File: t/check_vcfanno_toml.t [hunk 8]
```diff
@@ -100,11 +102,14 @@ write_toml(
     }
 );
 
+my %acctive_parameter;
```
Comment (henrikstranneheim) [inline t/check_vcfanno_toml.t:105]:
```suggestion
my %active_parameter;
```

File: t/check_vcfanno_toml.t [hunk 9]
```diff
@@ -100,11 +102,14 @@ write_toml(
...
+
 ## Given a toml config file with a file path
 my $is_ok = check_vcfanno_toml(
     {
-        parameter_name    => q{vta_vcfanno_config},
-        vcfanno_file_toml => $test_vta_vcfanno_config,
+        active_parameter_href => \%acctive_parameter,
```
Comment (henrikstranneheim) [inline t/check_vcfanno_toml.t:110]:
```suggestion
        active_parameter_href => \%active_parameter,
```

File: t/check_vcfanno_toml.t [hunk 10]
```diff
@@ -121,8 +126,9 @@ my $faulty_vta_vcfanno_config_file = catfile( $Bin,
 trap {
     check_vcfanno_toml(
         {
-            parameter_name    => q{vta_vcfanno_config},
-            vcfanno_file_toml => $faulty_vta_vcfanno_config_file,
+            active_parameter_href => \%acctive_parameter,
```
Comment (henrikstranneheim) [inline t/check_vcfanno_toml.t:129]:
```suggestion
            active_parameter_href => \%active_parameter,
```

File: t/data/references/rank_model_-v1.29-.ini [hunk 11]
```diff
@@ -1,6 +1,6 @@
 [Version]
   version = 1.28
```
Comment (henrikstranneheim) [inline t/data/references/rank_model_-v1.29-.ini:2]:
```suggestion
  version = 1.29
```

File: templates/mip_rd_dna_panel_config.yaml [hunk 12]
```diff
@@ -19,11 +19,11 @@ outdata_dir: cluster_constant_path!/case_id!/analysis_constant_path!
 outscript_dir: cluster_constant_path!/case_id!/analysis_constant_path!/scripts
 sample_info_file: cluster_constant_path!/case_id!/analysis_constant_path!/case_id!_qc_sample_info.yaml
 ## References
-vta_vcfanno_config: grch37_frequency_vcfanno_filter_config_-v1.4-.toml
+vta_vcfanno_config: grch37_frequency_vcfanno_filter_config_-v1.6-.toml
```
Comment (henrikstranneheim) [inline templates/mip_rd_dna_panel_config.yaml:22]: good catch!

File: templates/rank_model_-v1.29-.ini [hunk 13]
```diff
@@ -0,0 +1,973 @@
+[Version]
+  version = 1.28
```
Comment (henrikstranneheim) [inline templates/rank_model_-v1.29-.ini:2]:
```suggestion
  version = 1.29
```

File: lib/MIP/Vcfanno.pm (hunk 5 above)
Comment (jemten) [reply lib/MIP/Vcfanno.pm]: I've made some changes to this file. Should be clearer now but let me know what you think

File: lib/MIP/Vcfanno.pm [hunk 14]
```diff
@@ -32,26 +32,35 @@ sub check_vcfanno_toml {
 
 ## Function : Check that the supplied vcfanno toml config has mandatory keys and file exists for annotation array
 ## Returns  :
-## Arguments: $parameter_name    => Name of parameter
-##          : $vcfanno_file_toml => Toml config file
+## Arguments: $active_parameter_href => Holds all set parameter for analysis {REF}
+##          : $parameter_names_ref   => Name of parameters {REF}
```
Comment (henrikstranneheim) [inline lib/MIP/Vcfanno.pm:36]:
```suggestion
```

File: lib/MIP/Vcfanno.pm [hunk 15]
```diff
@@ -32,26 +32,35 @@ sub check_vcfanno_toml {
...
+##          : $vcfanno_file_toml     => Toml config file
```
Comment (henrikstranneheim) [inline lib/MIP/Vcfanno.pm:37]: add config and function to sub docs description

**Pull request merged**